mlrun/launcher/remote.py

```diff
@@ -90,9 +90,11 @@ def launch(
                 runtime.deploy(skip_deployed=True, show_on_failure=True)
 
             else:
-                raise mlrun.errors.MLRunRuntimeError(
-                    "Function image is not built/ready, set auto_build=True or use .deploy() method first"
-                )
+                if not runtime.spec.build.load_source_on_run:
```

Comment:
while this is a warning implying that we might fail, and suggest how to handle when it would fail - perhaps worth adding why we _dont_ raise an exception here and for which cases it is a valid flow

Reply:
I would suggest not to, leaving the server side to be the opinionated one.
perhaps only worth adding a comment that you "alerting the user and that it is up to the server to really validate it"

Reply:
So that's exactly what's being done here. I rephrased the message a bit to make it clear execution is about to fail.